openhands/server/routes/files.py

```diff
@@ -77,10 +77,16 @@ async def list_files(request: Request, path: str | None = None):
 
     file_list = [f for f in file_list if f not in FILES_TO_IGNORE]
 
-    async def filter_for_gitignore(file_list, base_path):
-        gitignore_path = os.path.join(base_path, '.gitignore')
+    async def has_gitignore() -> bool:
+        file_list = await call_sync_from_async(runtime.list_files, '')
```

Comment:
I wonder if this is a bit heavy, to do twice, for every click. (first at line 68)

Could we avoid it somehow? Or it doesn't matter?

Reply:
Yeah I agree we shouldn't do a `list_files` first--we should just try to access the file, and use error handling for FileNotFoundError

If that's causing logspam let's remove the logs upstream